.github/workflows/FreeBSD.yml

```diff
@@ -0,0 +1,250 @@
+name: Build Packages
+
+on:
+  push:
+    paths-ignore:
+    - 'doc/**'
+    - '**.md'
+    - 'appveyor.yml'
+    - '.travis.yml'
+  pull_request:
+    paths-ignore:
+    - 'doc/**'
+    - '**.md'
+    - 'appveyor.yml'
+    - '.travis.yml'
+
+env:
+  VERSION: 2024.2
+  THUNDER_RELEASE: ${{ startsWith(github.ref, 'refs/tags/2') }}
+  DEPS: git llvm17 libinotify openal-soft xorg 7-zip cmake ninja pkgconf ccache
+  QT: qt5-widgets qt5-core qt5-gamepad qt5-gui qt5-svg qt5-xml qmake qt5 
+  CC: clang17
+  CXX: clang++17
+  CCACHE_CFLAGS: -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
+  CCACHE_DIR: ".ccache"
+  CCACHE_COMPRESS: "true"
+  CCACHE_COMPRESSLEVEL: "6"
+  CCACHE_MAXSIZE: "400M"
+
+jobs:
+  freebsd14-1:
+    runs-on: ubuntu-latest
+    name: FreeBSD 14.1 CMake
+    steps:
+    - name: Prepare ccache timestamp
+      id: ccache_cache_timestamp
+      shell: cmake -P {0}
+      run: |
+        string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+        message("::set-output name=timestamp::${current_date}")
+    - name: Build FreeBSD 14.1 Packages
+      id: test1
+      uses: vmactions/freebsd-vm@v1
+      with:
+        envs: '
+          DEPS QT CC CXX
+          CCACHE_FLAGS
+          CCACHE_DIR
+          CCACHE_COMPRESS
+          CCACHE_COMPRESSLEVEL
+          CCACHE_MAXSIZE
+        '
+        usesh: true
+        sync: rsync
+        release: "14.1"
+        prepare: |
+          pkg install -y $DEPS
+          pkg install -y $QT
+        run: |
+          ccache -z
+          git clone --recursive --depth 1 https://github.com/thunder-engine/thunder
+          mkdir -pv tmp
+          cd tmp
+          CC=$CC CXX=$CXX cmake $CCACHE_FLAGS ../thunder -G "Ninja"
+          ninja -j 1
+          ninja install
+          mkdir -pv release
+          mv -v install-root release
+          7z a -t7z ../ThunderEngine-freebsd-14.1-x86_64.7z release/install-root
```

Comment:
I guess CI system will be stressed with so much targets for each commit

Reply:
> Do we need to keep all of BSD versions for the releases?

We don't have to. I can just keep the latest stable build and also make it only run one build at a time. Also, according to the [cache action](https://github.com/marketplace/actions/cache), we have 10GB of cache to work with per repository, so instead, I can set up individual caches for each platform and have them only fully build what's absolutely necessary, which would drastically reduce stress and could possibly be ported to the QBS build system to speed up iterations. What do you think about that?

Reply:
The BSD ones can be set to a larger size so adding Qt6 building would have some room since that will definitely take awhile but also would resolve some Qt concerns you brought up during #778 . Not sure how big the cache would need to be for that part yet, but I think there'd be a few GB left to work with.

Reply:
Update incoming. Might need to iterate again if it fails. But let's see how it turns out.

Reply:
This first build may take an hour or two to complete but if the caching code is correct (I used the example from the Cache runner repository after all and then modified everything else to fit with it), the cache here should grow to around 4 GB or so in size (or be 4 total 1 GB caches). Either way is perfect because having them timestamped with the date should allow for manual ejection of old cache files and the cache will automatically old ones once it hits the 10 GB cap.

Reply:
They may still need to be ran separately. :cry: Checking right now.

Reply:
I got this. I think we're still good. I just need to add an extra step. :joy_cat:

Reply:
Pushing updates to my fork to keep the runners accessible to you. I'm currently integrating a fork of the FreeBSD-vm runner into the system. It doesn't support running different BSD releases all on the same runner right now, so I added that in and might as well integrate it directly into the build pipeline.

See the progress here: https://github.com/bedwardly-down/freebsd-vm

Reply:
Caching doesn't work yet but the vms now do and take variables straight from the Github Runner.

Reply:
Reducing this down to the latest stable FreeBSD release for right now. Getting multiple builds and tests going is getting to be more trouble than it's worth at the moment. I can always return to it later. I have the foundation setup.

Getting the cache figured out and setting it for all of the supported platforms is more useful too.